File: frontend/app/src/main/java/com/example/commov/viewmodel/LoginUiState.java
package com.example.commov.viewmodel;

import androidx.annotation.StringRes;

public final class LoginUiState {
    public final String email;
    public final String password;
    public final boolean passwordVisible;
    @StringRes
    public final int emailErrorResId;
    @StringRes
    public final int passwordErrorResId;
    public final boolean loginAccepted;

    public LoginUiState(
            String email,
            String password,
            boolean passwordVisible,
            @StringRes int emailErrorResId,
            @StringRes int passwordErrorResId,
            boolean loginAccepted
    ) {
        this.email = email;
        this.password = password;
        this.passwordVisible = passwordVisible;
        this.emailErrorResId = emailErrorResId;
        this.passwordErrorResId = passwordErrorResId;
        this.loginAccepted = loginAccepted;
    }
}

File: frontend/app/src/main/java/com/example/commov/viewmodel/LoginUiState.kt
package com.example.commov.viewmodel

import androidx.annotation.StringRes

data class LoginUiState(
    val email: String,
    val password: String,
    val passwordVisible: Boolean,
    @StringRes val emailErrorResId: Int,
    @StringRes val passwordErrorResId: Int,
    @StringRes val generalErrorResId: Int,
    val isLoading: Boolean,
    val isCheckingSession: Boolean = false,
    val loginAccepted: Boolean
)
